### ISSUE TITLE
Type error: Cannot find module 'next/dist/shared/lib/app-router-context.shared-runtime' or its corresponding type declarations.

### ISSUE THREAD

**Issue by dungnk-admixion** [ndungtse/next13-progressbar#14]
**Describe the bug**
I'm running `next build` to build my Next.JS 13 app when this bug happens:

```
yarn run v1.22.19
$ next build
- info Loaded env from *:\***\.env.production
- info Creating an optimized production build
- info Compiled successfully
- info Linting and checking validity of types .Failed to compile.

./node_modules/next13-progressbar/dist/index.d.ts:3:73
Type error: Cannot find module 'next/dist/shared/lib/app-router-context.shared-runtime' or its corresponding type declarations.

  1 | import React from 'react';
  2 | import * as NProgress from 'nprogress';
> 3 | import * as next_dist_shared_lib_app_router_context_shared_runtime from 'next/dist/shared/lib/app-router-context.shared-runtime';
    |                                                                         ^
  4 | import { NavigateOptions } from 'next/dist/shared/lib/app-router-context.shared-runtime';
  5 |
  6 | interface NProviderProps extends Next13ProgressProps {
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**To Reproduce**
1. Install these dependencies in `package.json`:
```
"dependencies": {
	"@emotion/cache": "^11.11.0",
	"@emotion/react": "^11.11.3",
	"@emotion/styled": "^11.11.0",
	"@hookform/resolvers": "^3.3.3",
	"@mui/icons-material": "^5.15.2",
	"@mui/lab": "^5.0.0-alpha.158",
	"@mui/material": "^5.15.2",
	"@mui/material-nextjs": "^5.15.0",
	"@mui/system": "^5.15.2",
	"@mui/x-data-grid": "^6.18.6",
	"@types/google-libphonenumber": "^7.4.30",
	"@types/node": "^20.10.5",
	"@types/react": "^18.2.45",
	"@types/react-dom": "^18.2.18",
	"csstype": "^3.1.3",
	"eslint": "^8.56.0",
	"eslint-config-next": "^14.0.4",
	"google-libphonenumber": "^3.2.33",
	"jwt-decode": "^4.0.0",
	"next": "13.4.19",
	"next-recaptcha-v3": "^1.3.0",
	"next13-progressbar": "^1.1.3",
	"notistack": "^3.0.1",
	"react": "^18.2.0",
	"react-dom": "^18.2.0",
	"react-hook-form": "^7.49.2",
	"swr": "^2.2.4",
	"typescript": "^5.3.3",
	"typewriter-effect": "^2.21.0",
	"yup": "^1.3.3"
},
```
2. Run `next build`

**Expected behavior**
The build succeeds.

**Desktop (please complete the following information):**
 - OS: Windows 11 Pro
 - Browser:
 - Version:

**Additional context**
The build also fails on Google Cloud Build:
![image](https://github.com/ndungtse/next13-progressbar/assets/138103010/1b53cc0a-bd18-4079-9ba2-53014606260d)

**Comment by dungnk-admixion** [comment]
Anything on this yet @ndungtse ?

**Comment by ndungtse** [comment]
I the same dependences in a local project and it build successfully, I don't why yours is not working. 
Can you delete node_modules and reinstall them ? also delete .next

**Comment by dungnk-admixion** [comment]
I tried deleting both the `node_modules` and `.next` folder, and running `yarn` again to reinstall, but the problem persists (on my local machine). Please note that this also happens on Google Cloud Build, where every time my workflow runs, a fresh container is built with fresh `node_modules` and fresh `.next` folders.

When I access the file `node_modules\next13-progressbar\dist\index.d.ts` then the imports are indeed missing:
```
import * as next_dist_shared_lib_app_router_context_shared_runtime from 'next/dist/shared/lib/app-router-context.shared-runtime';
import { NavigateOptions } from 'next/dist/shared/lib/app-router-context.shared-runtime';
```
![image](https://github.com/ndungtse/next13-progressbar/assets/138103010/1b221e39-6f82-4c43-b6e2-025c5608c6dd)

**Comment by dungnk-admixion** [comment]
Can you share with me your `package.json` file? I'll see if I'm missing any dependencies/types... Thanks!

**Comment by dungnk-admixion** [comment]
@ndungtse I downgraded to `next13-progressbar@1.0.2` and everything is working again. I found that the problem starts from `1.1.0`. Can you see what changed from `1.0.2` to `1.1.0` that could have caused this problem?

**Comment by ndungtse** [comment]
I think the main change I made would be updating from next 13 to 14 in my dependencies, but this happened after v1.0.3. Check if v1.0.3 works then that should be the issue

**Comment by dungnk-admixion** [comment]
@ndungtse `1.0.3` works as expected.

**Comment by ndungtse** [comment]
Yeah, I think now issue is found. I'm going to work on it

**Comment by ndungtse** [comment]
@dungnk-admixion can you update to `1.2.0-alpha.1` to see if it works as expected?

**Comment by dungnk-admixion** [comment]
@ndungtse Nope. I installed with `yarn add next13-progressbar@1.2.0-alpha.1` and still got the error:
```
yarn run v1.22.19
$ next build
- info Loaded env from G:\admixion-landingpage\.env.production
- info Creating an optimized production build
- info Compiled successfully
- info Linting and checking validity of types .Failed to compile.

./node_modules/next13-progressbar/dist/index.d.ts:3:73
Type error: Cannot find module 'next/dist/shared/lib/app-router-context.shared-runtime' or its corresponding type declarations.

  1 | import React from 'react';
  2 | import * as NProgress from 'nprogress';
> 3 | import * as next_dist_shared_lib_app_router_context_shared_runtime from 'next/dist/shared/lib/app-router-context.shared-runtime';
    |                                                                         ^
  4 | import { NavigateOptions } from 'next/dist/shared/lib/app-router-context.shared-runtime';
  5 |
  6 | interface NProviderProps extends Next13ProgressProps {
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

**Comment by ndungtse** [comment]
> next_dist_shared_lib_app_router_context_shared_runtime

Can you try updating next to 14. to see if it works cause in that alpha version i downgraded next but it seems like it might not be the issue. @dungnk-admixion

**Comment by ndungtse** [comment]
> > next_dist_shared_lib_app_router_context_shared_runtime
> 
> Can you try updating next to 14. to see if it works cause in that alpha version i downgraded next but it seems like it might not be the issue. @dungnk-admixion

or check if the node_modules really got updatedby checking version in `node_modules/next13-progressbar/package.json`

**Comment by dungnk-admixion** [comment]
![image](https://github.com/ndungtse/next13-progressbar/assets/138103010/5a9739bd-34c3-4d6d-a2d5-4b70fe6bc687)
The `node_modules/next13-progressbar/package.json` file is showing the correct version `1.2.0-alpha.1`. Also I cannot upgrade to Next 14 as it is not yet stable, and introduces some breaking changes.

**Comment by ndungtse** [comment]
Just upgrade to see if it works and then downgrade again

**Comment by dungnk-admixion** [comment]
I have tried `next build` with these two combinations:
- Next `14.0.4` & next13-progressbar `1.1.3`
- Next `14.0.4` & next13-progressbar `1.2.0-alpha.1`

Both built successfully.

**Comment by ndungtse** [comment]
> I have tried `next build` with these two combinations:
> 
> * Next `14.0.4` & next13-progressbar `1.1.3`
> * Next `14.0.4` & next13-progressbar `1.2.0-alpha.1`
> 
> Both built successfully.

It's quite challenging as using your dependences and `v1.1.3` project is building successfuly
<img width="1572" alt="image" src="https://github.com/ndungtse/next13-progressbar/assets/95475354/5bbefd7b-1e47-4599-aad7-1b7ba61cb3d8">

**Comment by dungnk-admixion** [comment]
I checked your list of dependencies and noticed that your `eslint-config-next` is at `13.4.19`. I tried installing the same version of `eslint-config-next` and now my Next `13.4.19` app has been built correctly with the latest version of next13-progressbar `1.1.3`. What a weird error. Thanks for sharing your `package.json` file so I could compare with mine.

I think the issue has been solved. Thank you @ndungtse for your time. Great work as always!
